enterprise/kernel/src/main/java/org/neo4j/kernel/impl/store/format/highlimit/Reference.java

```diff
@@ -189,19 +197,17 @@ private static int maxBits()
 
     public static <SOURCE> long decode( SOURCE source, DataAdapter<SOURCE> adapter )
     {
-        PageCursor cursor = (PageCursor) source;
-
-        int header = cursor.getByte() & 0xFF;
+        int header = adapter.getByte( source ) & 0xFF;
         int sizeMarks = Integer.numberOfLeadingZeros( (~(header & 0xF8)) & 0xFF ) - 24;
         int signShift = 8 - sizeMarks - (sizeMarks == 5 ? 1 : 2);
         long signBit = ~((header >>> signShift) & 1) + 1;
         long register = (header & ((1 << signShift) - 1)) << 16;
-        register += cursor.getShort() & 0xFFFFL; // 3 bytes
+        register += adapter.getShort( source ) & 0xFFFFL; // 3 bytes
```

Comment:
This `getShort` call causes a slightly awkward `getShort` to be added to the adapter. Does it have a performance difference? Otherwise can we replace it with two `getByte` calls right in here instead (something which the adapter has to do anyway)?